connectors/steps/redshift/connect.py

```diff
@@ -0,0 +1,68 @@
+import json
+
+import redshift_connector
+import requests
+
+
+def get_credentials(imds_version: int, vh_worker_role: str):
+    aws_metadata_ip = "169.254.169.254"
+    if imds_version == 1:
+        url = f"http://{aws_metadata_ip}/latest/meta-data/iam/security-credentials/{vh_worker_role}"
+        credentials_response = requests.get(url, timeout=30)
+        credentials = json.loads(credentials_response.text)
+    elif imds_version == 2:
+        token_url = f"http://{aws_metadata_ip}/latest/api/token"
+        token_headers = {"X-aws-ec2-metadata-token-ttl-seconds": "60"}
+        token_response = requests.put(token_url, headers=token_headers, timeout=30)
+        token = token_response.text
+        role_url = (
+            f"http://{aws_metadata_ip}/latest/meta-data/iam/security-credentials/"
+        )
+        role_headers = {"X-aws-ec2-metadata-token": token}
+        role_response = requests.get(role_url, headers=role_headers, timeout=30)
+        role_name = role_response.text
+        credentials_url = f"{role_url}{role_name}"
+        credentials_response = requests.get(
+            credentials_url, headers=role_headers, timeout=30
+        )
+        credentials = credentials_response.json()
```

Comment:
Based on [what Botocore does](https://github.com/boto/botocore/blob/b9ff18918d8c4970dcc57cf867f31b8d1197a7c5/botocore/utils.py#L445-L463) I think we don't need to necessarily configure the IMDS version – we'd just try the token endpoint first, and if it returns a token, we use it, and if it returns 404/403/405, we assume we're on IMDSv1 and don't need it.

IOW, something like (untested, etc., ymmv)
```python
import requests


def get_imds_auth_headers(session):
    token_resp = session.put(
        url="http://169.254.169.254/latest/api/token",
        headers={"X-aws-ec2-metadata-token-ttl-seconds": "60"},
        timeout=5,
    )
    if token_resp.status_code in (403, 404, 405):  # IMDSv1
        return {}  # no auth headers
    elif token_resp.status_code == 200:
        return {"X-aws-ec2-metadata-token": token_resp.text}
    token_resp.raise_for_status()


def get_aws_credentials():
    with requests.Session() as session:
        req_headers = get_imds_auth_headers(session)
        role_resp = session.get(
            url="http://169.254.169.254/latest/meta-data/iam/security-credentials/",
            headers=req_headers,
            timeout=5,
        )
        role_resp.raise_for_status()
        role_name = role_resp.text
        credentials_resp = session.get(
            url=f"http://169.254.169.254/latest/meta-data/iam/security-credentials/{role_name}",
            headers=req_headers,
            timeout=5,
        )
        credentials_resp.raise_for_status()
        return credentials_resp.json()
```